packages/components/src/utils/mergeClassNames.ts

```diff
@@ -0,0 +1,27 @@
+import classnames from "classnames"
+
+type ClassNameValue = string | number | boolean | undefined | null
+
+export const mergeClassNames = <
```

Comment:
Trying to understand why this was created over using classnames?

Reply:
it was because RAC has [handy feature](https://react-spectrum.adobe.com/react-aria/styling.html#render-props) where you can pass function that receives state to `classname`. This is super useful when working with tailwind and we want to keep it. And because of that we need a helper that can deal correctly with the function call and merging.
I'll add a comment here to explain it for anyone coming to this later